packages/design-system/src/components/Field/Field.tsx

```diff
@@ -154,13 +155,27 @@ const Input = React.forwardRef<HTMLInputElement, InputProps>(
     }
 
     const hasError = Boolean(error);
+    const endActionRef = React.useRef<HTMLDivElement>(null);
+    const inputElementRef = React.useRef<HTMLInputElement>(null);
+    const inputRef = useComposedRefs(inputElementRef, ref);
 
     const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
       if (!disabled && onChange) {
         onChange(e);
       }
     };
 
+    React.useEffect(() => {
```

Comment:
I think this is a good use case for useLayoutEffect instead of useEffect, see this guide: https://react.dev/reference/react/useLayoutEffect#measuring-layout-before-the-browser-repaints-the-screen

So that we can have the right width from the very first render